client/src/api/block_builder/input_selection/core/requirement/amount.rs

```diff
@@ -218,7 +223,30 @@ impl InputSelection {
                 }
             }
 
-            // 4. Other kinds of outputs
+            // 4. Basic with other kind of address and multiple unlock conditions
+            {
+                log::debug!("Trying basic outputs with other types of address and multiple unlock conditions");
+
+                let inputs = self.available_inputs.iter().filter(|input| {
+                    if let Output::Basic(output) = &input.output {
+                        !output.address().is_ed25519() && output.unlock_conditions().len() > 1
+                    } else {
+                        false
+                    }
+                });
+
+                for input in inputs {
+                    inputs_sum += input.output.amount();
+                    newly_selected_inputs.push((input.clone(), None));
+                    newly_selected_ids.insert(*input.output_id());
+
```

Comment:
Do we also need to do extra storage deposit return stuff like in `// 2. Basic with ED25519 address and unexpired SDRUC` here?